### DISCUSSION TITLE
Welcome to Vigotech-VG-L7X Discussions!

### DISCUSSION THREAD

**Discussion by arkypita** [arkypita/Vigotech-VG-L7X#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by leroylee13** [comment]
I recently purchased a VG-L7X and have used LaserGRBL almost exclusively while learning the capabilities and limitations of my cutter.  I have been watching the discussions on the Custom Firmware Development thread and I'm excited by the possibility that I may be able to use Lightburn with my cutter.  I have a few questions about the firmware process.
1. Is the custom firmware ready to use?  I realize it is still being tweaked, but can I upload it to my machine and run Lightburn now?
2. I am a bit confused of the flashing procedure - in the directions on flashing ESP32 firmware, it states " is not possible to program the chip directly via the USB port without a little board modification-hack"which is connecting GPIO0 pin and the other end on GND, but that appears to be in conflict with the Readme file of the software hack that states you have to hold button 2 during the whole flashing process.
3. which button is button 1 and button 2?
Thank you guys for putting so much effort to make our machines easier to use!

**Reply by smoki3** [reply]
Hi leroylee,

1. The uploaded firmware is ready to use from my point of view. It is only tested by myself but I did some engraving job with lightburn and it was working fine. I actually just try to implement additional features
2. So this have to be adjusted. Arkypita started this project without the laser and just test setup. You don't need to wire anything else. Just follow the readme in the Custom Firmware folder
3. Button 1 is next to the power button (middle one) and button 2 the right one.

So please share your feedback

**Comment by arkypita** [comment]
It would be nice if someone record a video of the flash procedure. :-)

**Comment by leroylee13** [comment]
Can limit switches be added to the VG-LX7?
One thing that would significantly improve use of this laser engraver is the addition of limit switches.  I see that there are unused connectors along the edge of the circuit board which resemble connectors on some 3d printers which are used for limit switches - are these connector pins connected to I/O pins on the esp32 on the VG-LX7 and could be used for this function?

**Reply by smoki3** [reply]
I also thought about it. If I saw it right this connectors goes directly to the 328p not to the esp. But I think to use them as end stop switches they musst be connected to the 328p and the Grbl firmware must be adjusted.

Is it possible to configure end stops switched on the fly in Grbl?

**Reply by smoki3** [reply]
So as the connectors are labeled with X-L X-L which could be X-Limit and Y-Limit you connect a switch to it and enable the homing in Grbl with $22=1. 

I assume that it is working already

**Reply by MartinGiersberg** [reply]
This however yields the message "Grbl '$' system command was not recognized or supported." 
Any ideas on how to solve this?

**Reply by darjaallora** [reply]
Did anyone get these limit switches running?

**Reply by leroylee13** [reply]
I did not pursue it because I could not find a solid place to mount the end stops.  My LX7 is not mounted to a base so position does not stay consistent.

**Reply by darjaallora** [reply]
I have ID'd the jacks (JST PH 2.0), located some brackets for mounting the switches (https://www.thingiverse.com/thing:1794746)  ordered the switches and installed them Normal Open, per the GRBL docs I found on limit switches.  The problem now seems to be that $20 to $23 are removed from the GRBL firmware.  This is where my expertise ends.  How do I enable these switches in the firmware?  LaserGRBL's utility for changing the settings fails (invalid setting) because the pins seem to be undefined.  Assuming that I ID the pins correctly, how would I set those values and get that into the machine?

**Reply by darjaallora** [reply]
If someone would correct me if I am wrong, here is my current understanding of the situation.  The custom firmware offered here is firmware for the ESP32 that is gatekeeping the AVR chip where the GRBL firmware lives, and in order to activate the limit switches we need to replace the firmware that resides on the AVR chip?

**Comment by Lisker99** [comment]
I am trying to comple the code, but I get an error :
Arduino: 1.8.13 (Windows Store 1.8.42.0) (Windows 10), Board: "NodeMCU 1.0 (ESP-12E Module), 80 MHz, Flash, Legacy (new can return nullptr), All SSL ciphers (most compatible), 4MB (FS:2MB OTA:~1019KB), 2, v2 Lower Memory, Disabled, None, Only Sketch, 115200"
CustomFirmware-VG-L7x:5:21: fatal error: esp_wps.h: No such file or directory
 #include "esp_wps.h"
esp_wps.h: No such file or directory
Am I missing a library?
![esp error](https://user-images.githubusercontent.com/78666221/107131180-e3703a00-6899-11eb-86fb-298f6bda204b.JPG)

**Reply by Lisker99** [reply]
I got it to compile ok, but had to follow the directions here to get the right board group:
https://github.com/espressif/arduino-esp32/blob/master/docs/arduino-ide/boards_manager.md

I had originally done a search for NodeMCU Hardware library and added one that didn't have esp_wps apparently. Tried this one first, https://github.com/nodemcu/nodemcu-firmware , no dice....

Can that be added to the readme page for future users?

**Reply by Lisker99** [reply]
HUGE THANK YOU TO YOU GUYS!!! I know very little about programming, got through it and it is now communicating with LaserGRBL v4.2.1 like a charm!

One note for the ReadMe, though... if you could add to it that you release the right button as soon as the red light comes on it would help.  I held it too long the first 2 times (blue light came on) and it would not connect.  It would beep when IDE tried to connect but not let it go through.  The third try I let go of the right button before the blue light came on and it went through no problem.

**Reply by smoki3** [reply]
Okay cool!

I will update the readme a bit more. And add some screenshots how to flash.

I was not aware that holding the right button too long cause any problem
 On my device I can hold it the complete flashing process.

**Reply by Lisker99** [reply]
![Uploading 20210206_172049.jpg…]()
![Uploading 20210206_172252.jpg…]()
![Uploading 20210206_172610.jpg…]()
I took these while doing it if they help you...

**Comment by leroylee13** [comment]
I have been using my VG-L7X for nearly a year now with the firmware upgrade and use LightBurn almost exclusively.  The firmware made my machine go from barely usable to being able to do almost anything I can think of.  Thanks so much for sharing the firmware upgrades!

**Comment by Brainbug01** [comment]
![image](https://user-images.githubusercontent.com/35106961/148770648-d4fd7fb8-abac-4e20-8ad9-b7e82592e590.png)

https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_index.json

![image](https://user-images.githubusercontent.com/35106961/148770336-ad5d4709-ed1b-4110-a187-89e9d2c229b0.png)
Version 1.0.5

![image](https://user-images.githubusercontent.com/35106961/148770516-ebb282a8-2d70-46dd-a596-e9172a349924.png)
WifiManager

before the cable was connected to the PC, I had to start boot mode

**Comment by p10driver** [comment]
Hello, can i also flash the L5 4050 without wifi?

Greetings
